File: backend/src/main/java/com/park/utmstack/service/dto/threat_management/Adversary.java
package com.park.utmstack.service.dto.threat_management;

import com.park.utmstack.domain.shared_types.alert.Side;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Adversary {
    Side adversary;
}

File: backend/modules/alerts/dto/adversary.go
package dto

import "github.com/utmstack/utmstack/backend/modules/alerts/domain"

type AdversaryResponse struct {
	Adversary *domain.Side        `json:"adversary"`
	Alerts    []AlertWithChildren `json:"alerts"`
}

type AlertWithChildren struct {
	Alert    domain.UtmAlert   `json:"alert"`
	Children []domain.UtmAlert `json:"children"`
}

type AdversaryWrapper struct {
	Adversary *domain.Side `json:"adversary"`
}
